### PR TITLE
Adding initial version of image-builder tool and repo format

### PR DESCRIPTION
**Pull request by amitschang**
Introduces a new automatable approach to building SciServer images that aims to balance the need for interfacing updates
with the need for a stable environment for scientific use-cases, in a standardized manner.

**Notes**: 
* please see the rendered readme file in branch for some background and information on what the build tool is supposed to do and how to use it. 
* In this version, all image definitions are expected to be in the one repo. In the future this can be extended if needed.
* This commit also includes a nist image to serve as an example, which is basically sciserver-essentials with a modified startup file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,157 @@
...
+image, so that compat updates can be deployed even when there are no underlying science changes.
+
+# Setting up images
+
+Each science and compat directory should have a `sciserver-image.json` file describing details about the image and how
+to create it. For **science** it looks like:
+
+```json
+{
+  # image is the is the common name of the resultant image
```
Comment (esabol) [inline README.md:61]: This comment doesn't make any sense. Could you revise it?

Comment (amitschang) [reply README.md]: updated, I hope the new one provides a better description. Thanks!

Comment (esabol) [reply README.md]: Yes, thanks!

File: README.md [hunk 2]
```diff
@@ -1 +1,166 @@
...
+  # consumers)
+  "version": "2.4",
+
+  # The target refers to the build artifact name given to this particular
+  # science layer of image. Since multiple files can refer to the same image and
+  # version, this uniquely identifies this layer among all. It defaults to the
+  # current directory name. This name forms a tag
+  # (`{repo}/sci/{target}:{version}`) that can be used in the FROM line in other
+  # layers to build up layers
+  "target": "essentails-2021-10-12",
```
Comment (glemson) [inline README.md:79]: typo, essentails -> essentials

**Review by dmedv — APPROVED** [review]

**Comment by amitschang** [comment]
I think this has been here long enough, unless there are anymore comments, I think I will merge it in soon and then start to setup the automated build jobs, in addition to doing some migration for "sciserver essentials". This will help inform any necessary additions

**Pull request merged**